snap/validate.go

```diff
@@ -149,6 +237,14 @@ func validateField(name, cont string, whitelist *regexp.Regexp) error {
 	return nil
 }
 
+func validateAppSocket(socket *SocketInfo) error {
+	if err := ValidateAppSocketName(socket.Name); err != nil {
+		return err
+	}
+
+	return ValidateAppSocketListenAddress(socket, "listen-stream", socket.ListenStream)
```

Comment:
what's the driver for passing `"listen-stream"` in like this? It's making everything more complex and I'm not seeing the upside.

Reply:
The field is passed in so that the error message can mention it. The idea is to reuse the code for other socket types (e.g. `"listen-datagram"`)